Libraries/LibWeb/DOM/Document.cpp

```diff
@@ -6339,6 +6340,181 @@ bool Document::fullscreen_enabled() const
     return true;
 }
 
+void Document::exit_fullscreen_fully()
+{
+    // 1. If document’s fullscreen element is null, terminate these steps.
+    GC::Ptr<Element> fullscreened_element = fullscreen_element();
+    if (!fullscreened_element)
+        return;
+
+    // 2. Unfullscreen elements whose fullscreen flag is set, within document’s top layer, except for document’s fullscreen element.
+    Vector<GC::Ref<Element>, 8> fullscreen_elements;
+    for (auto const& element : top_layer_elements()) {
+        if (element->is_fullscreen_element() && element != fullscreened_element)
+            fullscreen_elements.append(element);
+    }
+
+    for (auto const& element : fullscreen_elements) {
+        unfullscreen_element(element);
+    }
+
+    HTML::TemporaryExecutionContext context(relevant_settings_object().realm(), HTML::TemporaryExecutionContext::CallbacksEnabled::Yes);
```

Comment:
why is this necessary? the caller should take care of this.

Reply:
Will be removed (as discussed on discord).